boot/mbed/mcuboot_imgtool.cmake

```diff
@@ -0,0 +1,115 @@
+    endif()
+endif()
+
+# Signing key -------------------------------------------------------------------------------------
+
+set(MCUBOOT_SIGNING_KEY "" CACHE STRING "Path to key file (.pem) used to sign firmware updates for your device.  This file must be kept safe!")
+
+# Make sure the signing key path is absolute for EXISTS, relative to the top level build dir
+get_filename_component(MCUBOOT_SIGNING_KEY_ABSPATH "${MCUBOOT_SIGNING_KEY}" ABSOLUTE BASE_DIR ${CMAKE_BINARY_DIR})
+
+if("${MCUBOOT_SIGNING_KEY}" STREQUAL "" OR NOT EXISTS "${MCUBOOT_SIGNING_KEY_ABSPATH}")
+    message(FATAL_ERROR "Must specify path to valid image signing key via MCUBOOT_SIGNING_KEY in order to build this project.")
+endif()
+
+# Imgtool usage functions --------------------------------------------------------------------------
+
+#
+# Generate an initial image hex file for the given executable target.
+# This initial image is what should be flashed to a blank device (along with the bootloader).
+#
+# NOTE: This function must be called *after* mbed_set_post_build() for the target!
+#
+#
+# If you wish to specify the output path of the initial image, pass that as the second argument to
+# this function.  Otherwise, it will default to ${CMAKE_CURRENT_BINARY_DIR}/<target name>-initial-image.hex
+#
+function(mcuboot_generate_initial_image TARGET) # optional 2nd arg: file name
+
+    # Figure out file name
+    if("${ARGN}" STREQUAL "")
+        set(INITIAL_IMAGE_PATH ${CMAKE_CURRENT_BINARY_DIR}/$<TARGET_FILE_BASE_NAME:${TARGET}>-initial-image.hex)
+    else()
+        set(INITIAL_IMAGE_PATH ${ARGN})
+    endif()
+
+    if("${MBED_OUTPUT_EXT}" STREQUAL "bin")
+        message(FATAL_ERROR "Hex file output must be enabled to use mcuboot.  Set MBED_OUTPUT_EXT to empty string or to 'hex' in your top level CMakeLists.txt!")
+    endif()
+
+    if("${PROJECT_VERSION}" STREQUAL "")
+        message(FATAL_ERROR "You must set the project version to sign images by passing a version number into your app's project() command!")
+    endif()
+
+    # mbed_generate_bin_hex() puts the hex file at the following path
+    set(TARGET_HEX_FILE ${CMAKE_CURRENT_BINARY_DIR}/$<TARGET_FILE_BASE_NAME:${TARGET}>.hex)
+
+    # Grab header size
+    if(NOT "${MBED_CONFIG_DEFINITIONS}" MATCHES "MCUBOOT_HEADER_SIZE=(0x[0-9A-Fa-f]+)")
+        message(FATAL_ERROR "Couldn't find MCUBOOT_HEADER_SIZE in Mbed configuration!")
+    endif()
+    set(HEADER_SIZE_HEX ${CMAKE_MATCH_1})
+
+    # Grab slot size
+    if(NOT "${MBED_CONFIG_DEFINITIONS}" MATCHES "MCUBOOT_SLOT_SIZE=(0x[0-9A-Fa-f]+)")
+        message(FATAL_ERROR "Couldn't find MCUBOOT_SLOT_SIZE in Mbed configuration!")
+    endif()
+    set(SLOT_SIZE_HEX ${CMAKE_MATCH_1})
+
+    add_custom_command(
+        TARGET ${TARGET}
+        POST_BUILD
+        COMMAND
+            ${Python3_EXECUTABLE} -m imgtool.main
+            sign
+
```

Comment:
empty lines here should be removed, this is all one command

Reply:
The empty lines here are important for grouping the comments with the lines they are commenting

Reply:
It's not in-fitting with the project e.g. https://github.com/mcu-tools/mcuboot/blob/main/boot/espressif/CMakeLists.txt#L407 and https://github.com/mcu-tools/mcuboot/blob/main/boot/zephyr/CMakeLists.txt#L372 so should be updated to match

Reply:
removed blanks